### PR TITLE
Calypso build: Configurable WP deps

### PR DESCRIPTION
**Pull request by sirreal**
Calypso-build always externalizes WP deps. Allow configuration.

This adds `WP` webpack env to be defined to enable WP externals.

## Questions

This is the first introduction of `env` usage to the `calypso-build` webpack config. They appear to be the way to configure arbitrary behavior from the webpack CLI.

`WP` - is this a good name for targeting the WordPress environment?

## Testing instructions

`npx lerna run build` -> WP deps will be external:

```
@automattic/o2-blocks: Built at: 04/04/2019 8:23:23 PM
@automattic/o2-blocks:          Asset      Size  Chunks             Chunk Names
@automattic/o2-blocks:     editor.css  3.03 KiB  editor  [emitted]  editor
@automattic/o2-blocks:      editor.js  49.6 KiB  editor  [emitted]  editor
@automattic/o2-blocks: editor.rtl.css  3.01 KiB  editor  [emitted]  editor
@automattic/o2-blocks: Entrypoint editor = editor.css editor.js editor.rtl.css
@automattic/o2-blocks: [../o2-blocks/src/editor-notes/editor.js] 1.14 KiB {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/editor-notes/editor.scss] 39 bytes {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/editor.js] 117 bytes {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/prev-next/editor.js] 2.36 KiB {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/prev-next/editor.scss] 39 bytes {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/todo/editor.js] 9.54 KiB {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/todo/editor.scss] 39 bytes {editor} [built]
@automattic/o2-blocks: [../o2-blocks/src/todo/item.js] 3.68 KiB {editor} [built]
@automattic/o2-blocks: [@wordpress/blocks] external "wp.blocks" 42 bytes {editor} [built]
@automattic/o2-blocks: [@wordpress/components] external "wp.components" 42 bytes {editor} [built]
@automattic/o2-blocks: [@wordpress/editor] external "wp.editor" 42 bytes {editor} [built]
@automattic/o2-blocks: [@wordpress/element] external "wp.element" 42 bytes {editor} [built]
@automattic/o2-blocks: [@wordpress/i18n] external "wp.i18n" 42 bytes {editor} [built]
```

### CODE REVIEW

**Comment by simison** [comment]
> WP - is this a good name for targeting the WordPress environment?

Personally I try to always avoid abbreviations but I think this is fine, considering Gutenberg also uses `WP_BUNDLE_ANALYZER`, `WP_DEVTOOL` etc so `WP` aligns well.

**Comment by simison** [comment]
Conceptually seems fine but needs a rebase.

**Comment by sirreal** [comment]
Rebased

**Comment by simison** [comment]
Needs rebase with `/apps` folder now.

**Comment by simison** [comment]
Will `npx @automattic/calypso-build --env.WP='true'` work, too?

**Comment by sirreal** [comment]
Rebased

> Will `npx @automattic/calypso-build --env.WP='true'` work, too?

I don't think so, we'd need to alias calypso-build to webpack, right now its bin points at build-package:

https://github.com/Automattic/wp-calypso/blob/2219904170e3666b564e1c7f08446f0d4f02f67f/packages/calypso-build/package.json#L24

File: apps/o2-blocks/package.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
 	"scripts": {
 		"clean": "npx rimraf dist",
 		"prebuild": "npm run clean",
-		"build": "webpack --config='../../packages/calypso-build/webpack.config.js' --output-path='./dist' editor='./src/editor.js' "
+		"build": "webpack --config='../calypso-build/webpack.config.js' --output-path='./dist' --env.WP='true' editor='./src/editor.js' "
```
Comment (simison) [inline apps/o2-blocks/package.json]: Should prolly be `../packages/calypso-build` here

Comment (sirreal) [reply apps/o2-blocks/package.json]: 😩 Everything got mixed up when I spotted the problem in #32138 

I think it should be in order now.

File: packages/calypso-build/webpack.config.js [hunk 2]
```diff
@@ -107,7 +107,7 @@ function getWebpackConfig(
 			new webpack.IgnorePlugin( /^\.\/locale$/, /moment$/ ),
 			...SassConfig.plugins( { cssFilename, minify: ! isDevelopment } ),
 			new DuplicatePackageCheckerPlugin(),
-			new WordPressExternalDependenciesPlugin(),
+			...( env.WP ? [ new WordPressExternalDependenciesPlugin() ] : [] ),
```
Comment (simison) [inline packages/calypso-build/webpack.config.js]: Because of this change, I suppose we need to add `WP` env in Jetpack before publishing the next version?

Comment (sirreal) [reply packages/calypso-build/webpack.config.js]: Yep! Breaking change!

**Comment by sirreal** [comment]
I added a bump to beta for `calypso-build` here (fyi @ockham )

**Pull request merged**